pkg/webhook/authentication/http.go

```diff
@@ -34,6 +34,20 @@ import (
 var authenticationScheme = runtime.NewScheme()
 var authenticationCodecs = serializer.NewCodecFactory(authenticationScheme)
 
+// adapted from https://github.com/kubernetes/kubernetes/blob/c28c2009181fcc44c5f6b47e10e62dacf53e4da0/staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go
+//
+// From https://github.com/kubernetes/apiserver/blob/d6876a0600de06fef75968c4641c64d7da499f25/pkg/server/config.go#L433-L442C5:
+//
+//	     1.5MB is the recommended client request size in byte
+//		 the etcd server should accept. See
+//		 https://github.com/etcd-io/etcd/blob/release-3.4/embed/config.go#L56.
+//		 A request body might be encoded in json, and is converted to
+//		 proto when persisted in etcd, so we allow 2x as the largest request
+//		 body size to be accepted and decoded in a write request.
```

Comment:
Q: What does this have to do with the TokenReview request?

Reply:
You are right, the TokenReview request does not come from etcd.
Not sure what a better upper-limit would be.
I did read somewhere that JWTs have a upper size limit because they have to be includable in the header of a HTTP request, maybe that is a starting point?

Reply:
After a quick look at google/stackoverflow it seems that neither JWT nor HTTP defines a max size for a JWT / a HTTP header. Just that there are limits that webservers have in place: https://stackoverflow.com/questions/686217/maximum-on-http-header-values

It seems usually HTTP header limits are between 4 and 48k. Maybe let's use 1 MB as a limit which should be low enough given what we're protecting against.

Then I would add a comment that the TokenReview mostly contains a bearer token which at most should have a few kb's of size, so we picked 1 MB to have plenty of buffer.

Reply:
@alvaroaleman @vincepri ^^ any other ideas on how to pick a limit?

Reply:
The logic above makes sense to me, in both cases I'd add a comment that if this happens, folks should open an issue, and link to https://github.com/kubernetes-sigs/controller-runtime/issues/new